### PR TITLE
Fix job test on AKS win and linux.

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Fix job test on AKS win and linux.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will partially addresses: #2982

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [X] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by yaron2 — APPROVED** [review]

**Review by wcs1only — DISMISSED** [review]

**Review by pkedy — DISMISSED** [review]

**Comment by artursouza** [comment]
Thanks for reviewing this. As expected, it failed on win cluster - although I did test on my private AKS win cluster too.

**Comment by yaron2** [comment]
@artursouza may the force be with you.

**Comment by artursouza** [comment]
@yaron2  @pkedy  @wcs1only  Need another review now. Please, take a look. job_test.go now passes on Linux and Windows. We still need another PR for windows E2E tests.

File: tests/apps/job-publisher/app.go
Comment (wcs1only) [inline tests/apps/job-publisher/app.go:64]: We talked about this on the call this morning, but commenting for the sake of posterity. Did we add this specifically so that our ValidateSidecar() function works correctly? Or will a user running jobs that take <1m experience some kind of failure state? If it is the latter, please make sure we have an issue so we don't lose track of it.

**Review by wcs1only — APPROVED** [review]
Approving so as to unblock the release, but please see my comment above and make sure we have an issue created, if we need one.

File: tests/apps/job-publisher/app.go
Comment (artursouza) [reply tests/apps/job-publisher/app.go:64]: Validation of sidecar is also made possible with this change. But, the reason is (like you said) to overcome an issue where K8s would not report the Job as completed on Windows.

Comment (artursouza) [reply tests/apps/job-publisher/app.go:64]: Issue: https://github.com/dapr/dapr/issues/2995

Comment (artursouza) [reply tests/apps/job-publisher/app.go:64]: I am actually thinking if this is a side effect of the Dapr sidecar being injected and then shutdown by the app and how K8s might interpret that as a failure. The shutdown code for sidecar does an exit(0), so it would be something related to timing maybe: https://github.com/dapr/dapr/blob/8fc2332dea0e5ec07333e24a942d7d54848c2c65/pkg/runtime/runtime.go#L1498

**Pull request merged**
